### PR TITLE
Better kubelet logging for probes

### PR DESCRIPTION
**Pull request by thockin**
Log when we actually run probes and event when they fail.  Print the output of a probe, too.

I tried to keep all the same semantics wrt status and errors, even where it is a bit weird (timeouts in particular).

### CODE REVIEW

**Comment by vmarmol** [comment]
Re balancing to @dchen1107 who volunteered :)

**Comment by thockin** [comment]
Closes #8179

File: pkg/client/kubelet.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ type KubeletClient interface {
 
 // KubeletHealthchecker is an interface for healthchecking kubelets
 type KubeletHealthChecker interface {
-	HealthCheck(host string) (probe.Result, error)
+	HealthCheck(host string) (probe.Result, string, error)
```
Comment (vmarmol) [inline pkg/client/kubelet.go]: nit: Can we document the return types?

File: pkg/kubelet/prober/prober.go [hunk 2]
```diff
@@ -92,84 +92,109 @@ func NewTestProber(
...
-		pb.readinessManager.SetReadiness(containerID, false)
-		return live, nil
+	if container.LivenessProbe != nil {
+		live, output, err := pb.probeLiveness(pod, status, container, containerID, createdAt)
+		if err != nil || live != probe.Success {
+			// Liveness failed in one way or another.
+			pb.readinessManager.SetReadiness(containerID, false)
+			ref, ok := pb.refManager.GetRef(containerID)
+			if !ok {
+				glog.Warningf("No ref for pod '%v' - '%v'", containerID, container.Name)
```
Comment (vmarmol) [inline pkg/kubelet/prober/prober.go]: nit: %q instead of %v

File: pkg/kubelet/prober/prober.go [hunk 3]
```diff
@@ -92,84 +92,109 @@ func NewTestProber(
...
+				return probe.Unknown, err
+			} else { // live != probe.Success
+				glog.V(1).Infof("Liveness probe for %q failed (%v): %s", ctrName, live, output)
+				if ok {
+					pb.recorder.Eventf(ref, "unhealthy", "Liveness probe failed: %s", output)
+				}
+				return live, nil
+			}
+		}
+		glog.V(1).Infof("Liveness probe for %q succeeded", ctrName)
```
Comment (vmarmol) [inline pkg/kubelet/prober/prober.go]: Do think V(1) is too chatty for this?

**Comment by vmarmol** [comment]
LGTM, mainly nits

File: pkg/client/kubelet.go (hunk 1 above)
Comment (thockin) [inline pkg/client/kubelet.go]: next push

File: pkg/kubelet/prober/prober.go (hunk 2 above)
Comment (thockin) [inline pkg/kubelet/prober/prober.go]: next push

File: pkg/kubelet/prober/prober.go (hunk 3 above)
Comment (thockin) [inline pkg/kubelet/prober/prober.go]: yes, moved to V(3)

**Comment by thockin** [comment]
new push is up

File: pkg/kubelet/prober/prober.go [hunk 4]
```diff
@@ -92,84 +92,109 @@ func NewTestProber(
...
+				return probe.Unknown, err
+			} else { // ready != probe.Success
+				glog.V(1).Infof("Readiness probe for %q failed (%v): %s", ctrName, ready, output)
+				if ok {
+					pb.recorder.Eventf(ref, "unhealthy", "Readiness probe failed: %s", output)
+				}
+				return ready, nil
+			}
+		}
+		glog.V(1).Infof("Readiness probe for %q succeeded", ctrName)
```
Comment (vmarmol) [inline pkg/kubelet/prober/prober.go]: Same here about V(3)?

**Comment by vmarmol** [comment]
LGTM, thanks @thockin! We can merge this when the freeze is removed.

File: pkg/kubelet/prober/prober.go (hunk 4 above)
Comment (thockin) [inline pkg/kubelet/prober/prober.go]: Sigh.  yes.

**Comment by dchen1107** [comment]
LGTM. Merging...

**Pull request merged**